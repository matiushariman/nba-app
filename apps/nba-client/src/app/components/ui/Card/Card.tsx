import type { CardProps } from './Card.types';

const Card = ({ children, className, ...rest }: CardProps) => (
  <div className={`${className} shadow-md bg-white rounded`} {...rest}>
    {children}
  </div>
);

export default Card;
