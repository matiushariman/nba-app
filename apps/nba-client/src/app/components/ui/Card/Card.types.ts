import type { ReactNode, HTMLAttributes } from 'react';

export interface CardProps extends HTMLAttributes<HTMLDivElement> {
  readonly children?: ReactNode;
}
