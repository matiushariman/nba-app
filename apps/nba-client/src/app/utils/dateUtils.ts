import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
