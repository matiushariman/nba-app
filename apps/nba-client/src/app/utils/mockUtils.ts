export const shouldUseMock = process.env['NODE_ENV'] === 'development';
