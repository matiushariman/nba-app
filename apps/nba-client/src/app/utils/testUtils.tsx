import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
