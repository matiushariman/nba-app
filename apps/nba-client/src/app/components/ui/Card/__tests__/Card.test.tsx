import Card from '..';
import { render } from '../../../../utils/testUtils';

describe('components/ui/Card', () => {
  it('should render Card that matches snapshot', () => {
    const { baseElement } = render(<Card>Hello</Card>);

    expect(baseElement).toMatchSnapshot();
  });
});
