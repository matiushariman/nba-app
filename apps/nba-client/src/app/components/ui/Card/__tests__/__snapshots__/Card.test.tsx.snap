// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/Card should render Card that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="undefined shadow-md bg-white rounded"
    >
      Hello
    </div>
  </div>
</body>
`;
